src/test/java/net/whitbeck/rdbparser/RdbParserTest.java

```diff
@@ -832,6 +841,57 @@ public void hashmapAsZipList() throws Exception {
     }
   }
 
+  @Test
+  public void hashmapAsListpack() throws Exception {
+    if (rdbVersion > 9) {
+      StringBuffer sb = new StringBuffer(16384);
+      for (int i=0; i<64; ++i) {
+        sb.append("x");
+      }
+      String mediumString = sb.toString();
+      for (int i=0; i<256; ++i) {
+        sb.append(mediumString);
+      }
+      String longString = sb.toString();
+      List<String> list = Arrays.asList(
+                                         "30", // 7 bit integer
+                                        "500", // 13 bit signed integer
+                                        "-30", // 13 bit signed integer
+                                        "16000", // 16 bit integer
+                                        "-16000", // 16 bit signed integer
+                                        "300000", // 24 bit integer
+                                        "-300000", // 24 bit signed integer
+                                        "30000000", // 32 bit integer
+                                        "-30000000", // 32 bit signed integer
+                                        "9000000000", // 64 bit integer
+                                        "-9000000000", // 64 bit signed integer
+                                        "loremipsum", // 6 bit string
+                                        mediumString, // 12 bit string
+                                        longString // 32 bit string
+      );
+      Map<String, String> map = new HashMap<String, String>();
+      for (int i=0; i<list.size(); ++i) {
+        map.put(Integer.toString(i), list.get(i));
+      }
+      jedis.configSet("hash-max-ziplist-value", Integer.toString(longString.length()));
```

Comment:
Is this setting still relevant?

Reply:
It does seem to work, but I changed it to `hash-max-listpack-value` in the new test in accordance with the documentation: https://redis.io/docs/management/optimization/memory-optimization/. Looks like it changes again in `7.2`, which is RDB 11.